coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -2,51 +2,50 @@
 
 angular.module('coalaHtmlApp')
   .controller('MainCtrl', function ($scope, $http) {
-    var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + '/coala.json';
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        var knownFiles = {};
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
-                    });
-              }
-              knownFiles[sourceRange.file].push({
-                "start":    sourceRange.start.line,
-                "end":      sourceRange.end.line,
-                "diffs":    el.diffs,
-                "message":  el.message,
-                "origin":   el.origin,
-                "severity": el.severity
+      var parseCoalaProject = function(projectDir, coalaJSON) {
+          $http.get(projectDir + '/file_data.json')
+            .success(function(file_data) {
+                $scope.file_dict = {};
+                Object.keys(file_data).forEach(function(file_name) {
+                    if (!(file_name in $scope.file_dict)){
+                      $scope.file_dict[file_name]=file_data[file_name];
+                    }
+                });
+              coalaJSON = coalaJSON || projectDir + '/coala.json';
+              $scope.data = [];
+              $http.get(coalaJSON).success(function(response){
+                  var knownFiles = {};
+                  var parseResult = function(el){
+                      el.affected_code.forEach(function(sourceRange){
+                          if (!(sourceRange.file in knownFiles)){
+                              knownFiles[sourceRange.file] = [];
+                          }
+                          knownFiles[sourceRange.file].push({
+                            "start":    sourceRange.start.line,
+                            "end":      sourceRange.end.line,
+                            "diffs":    el.diffs,
+                            "message":  el.message,
+                            "origin":   el.origin,
+                            "severity": el.severity
+                          });
+                      });
+                };
+                for (var errors in response.results) {
+                  if (response.results[errors][0]){ // If error exists
```

Comment:
`response` has `results` as key and errors are contents of the list. So a
list of errors
On 17 May 2016 23:06, "Attila" <notifications@github.com> wrote:

> In coalahtml/_coalahtml/app/controllers/main.js
> <https://github.com/coala-analyzer/coala-html/pull/46#discussion_r63567995>
> :
>
> > +                      el.affected_code.forEach(function(sourceRange){
> > +                          if (!(sourceRange.file in knownFiles)){
> > +                              knownFiles[sourceRange.file] = [];
> > +                          }
> > +                          knownFiles[sourceRange.file].push({
> > +                            "start":    sourceRange.start.line,
> > +                            "end":      sourceRange.end.line,
> > +                            "diffs":    el.diffs,
> > +                            "message":  el.message,
> > +                            "origin":   el.origin,
> > +                            "severity": el.severity
> > +                          });
> > +                      });
> > +                };
> > +                for (var errors in response.results) {
> > +                  if (response.results[errors][0]){ // If error exists
>
> So in response.results the keys are results, but you iterate over them
> and call them errors? Am I missing something?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-html/pull/46/files/e8c8c1fee8f3c25714d30f2460b5d41133583994..46a76f2b687a526bc095bdaa2950f834edf86d5d#r63567995>
>